### PR TITLE
[kjobctl] Add raycluster flag for RayJob mode.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add raycluster flag for RayJob mode on kjobctl.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by trasc — COMMENTED** [review]
overall LGTM, but I think we should a a column in the list command for this after 

- #2718

is merged

**Comment by mbobrovskyi** [comment]
> overall LGTM, but I think we should a a column in the list command for this after
> 
> * [[kjobctl] Add list RayJob command. #2718](https://github.com/kubernetes-sigs/kueue/pull/2718)
> 
> is merged

Good idea. Added.

File: cmd/experimental/kjobctl/pkg/cmd/list/list_test.go [hunk 1]
```diff
@@ -81,6 +81,7 @@ ns2         j2     profile2   lq1           3/3           60m        60m
 					Profile("profile1").
 					LocalQueue("lq1").
 					JobStatus(rayv1.JobStatusSucceeded).
+					RayClusterName("cq1").
```
Comment (trasc) [inline cmd/experimental/kjobctl/pkg/cmd/list/list_test.go:84]: nit
```suggestion
					RayClusterName("rc1").
```

File: cmd/experimental/kjobctl/pkg/cmd/list/list_test.go [hunk 2]
```diff
@@ -90,15 +91,16 @@ ns2         j2     profile2   lq1           3/3           60m        60m
 					Profile("profile2").
 					LocalQueue("lq2").
 					JobStatus(rayv1.JobStatusSucceeded).
+					RayClusterName("cq1").
```
Comment (trasc) [inline cmd/experimental/kjobctl/pkg/cmd/list/list_test.go:94]: maybe we should have one with no cluster set, (I think that while the job is suspended it's empty)

File: cmd/experimental/kjobctl/pkg/cmd/create/create.go [hunk 3]
```diff
@@ -208,6 +210,13 @@ var createModeSubcommands = map[string]modeSubcommand{
 				"MinReplicas denotes the minimum number of desired Pods for this worker group.")
 			subcmd.Flags().StringToIntVar(&o.MaxReplicas, maxReplicasFlagName, nil,
 				"MaxReplicas denotes the maximum number of desired Pods for this worker group, and the default value is maxInt32.")
+			subcmd.Flags().StringVar(&o.RayCluster, rayClusterFlagName, "",
+				"Ray cluster name where ray job will crate.")
```
Comment (trasc) [inline cmd/experimental/kjobctl/pkg/cmd/create/create.go:214]: can we reformulate this? 

"Existing cluster on which the job will be created" or something.

File: cmd/experimental/kjobctl/go.sum [hunk 4]
```diff
@@ -100,6 +100,8 @@ github.com/imdario/mergo v0.3.16 h1:wwQJbIsHYGMUyLSPrEq1CT16AhnhNJQ51+4fdHUnCl4=
 github.com/imdario/mergo v0.3.16/go.mod h1:WBLT9ZmE3lPoWsEzCh9LPo3TiwVN+ZKEjmz+hD27ysY=
 github.com/inconshreveable/mousetrap v1.1.0 h1:wN+x4NVGpMsO7ErUn/mUI3vEoE6Jt13X2s0bqwp9tc8=
 github.com/inconshreveable/mousetrap v1.1.0/go.mod h1:vpF70FUmC8bwa3OWnCshd2FqLfsEA9PFc4w1p2J65bw=
+github.com/jarcoal/httpmock v1.2.0 h1:gSvTxxFR/MEMfsGrvRbdfpRUMBStovlSRLw0Ep1bwwc=
+github.com/jarcoal/httpmock v1.2.0/go.mod h1:oCoTsnAz4+UoOUIf5lJOWV2QQIW5UoeUI6aM2YnWAZk=
```
Comment (trasc) [inline cmd/experimental/kjobctl/go.sum:104]: how do we have this with no go.mod updates?

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/go.sum:104]: It's a very big question for me too :)

File: cmd/experimental/kjobctl/pkg/cmd/list/list_test.go (hunk 1 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/cmd/list/list_test.go]: Thanks. Fixed.

File: cmd/experimental/kjobctl/pkg/cmd/list/list_test.go (hunk 2 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/cmd/list/list_test.go]: You right, it's a good point. Done.

File: cmd/experimental/kjobctl/pkg/cmd/create/create.go (hunk 3 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/pkg/cmd/create/create.go]: Fixed. Thanks.

File: cmd/experimental/kjobctl/go.sum (hunk 4 above)
Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/go.sum:104]: Nothing.

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/go.sum:104]: I added `rayutil` import. 

https://github.com/kubernetes-sigs/kueue/blob/2b225325a3521bf3df618265d9fd25524abab6e7/cmd/experimental/kjobctl/pkg/builder/ray_job_builder.go#L24

Inside this package using `github.com/jarcoal/httpmock`.

Comment (mbobrovskyi) [reply cmd/experimental/kjobctl/go.sum:104]: Maybe because I'm using only one constant from this package.

**Review by trasc — COMMENTED** [review]
/lgtm

**Pull request merged**